src/span.js

```diff
@@ -0,0 +1,220 @@
+     **/
+    context(): SpanContext {
+        return this._spanContext;
+    }
+
+    /**
+     * Returns the tracer associated with this span.
+        this._duration;
+     * @return {Tracer} - returns the tracer associated witht this span.
+     **/
+    tracer(): any {
+        return this._tracer;
+    }
+
+    /**
+     * Sets the operation name on this given span.
+     *
+     * @param {string} name - The name to use for setting a span's operation name.
+     **/
+    setOperationName(name: string): void {
+        this._name = name;
+    }
+
+    /**
+     * Finishes a span which has the effect of reporting it and
+     * setting the finishTime on the span.
+     *
+     * @param {number} finishTime - The time on which this span finished.
+     **/
+    finish(finishTime: number): void {
+        if (this._duration !== undefined) {
+            throw 'You can only call finish() on a span once.';
+        }
+
+        if (this._spanContext.isSampled()) {
+            let endTime = finishTime || Utils.getTimestampMicros();
+            this._duration = endTime - this._start;
+            this._tracer._report(this);
+        }
+    }
+
+    /**
+     * Adds a set of tags to a span.
+     *
+     * @param {Object} keyValuePairs - An object with key value pairs
+     * that represent tags to be added to this span.
+     **/
+    addTags(keyValuePairs: any): void {
+        if (this._spanContext.isSampled()) {
+            for (let key in keyValuePairs) {
+                if (keyValuePairs.hasOwnProperty(key)) {
+                    let value = keyValuePairs[key];
+                    let tag;
+                    if (typeof value === 'number' && (Math.floor(value) === value)) {
+                        tag = Utils.createIntegerTag(key, value);
+                    } else if (typeof value === 'boolean') {
+                        tag = Utils.createBooleanTag(key, value);
+                    } else {
+                        tag = Utils.createStringTag(key, value);
+                    }
+                    this._binaryAnnotations.push(tag);
+                }
+            }
+        }
+    }
+
+    /**
+     * Adds a log event, or payload to a span.
+     *
+     * @param {Object} fields - an object that represents the fields to log.
+     * @param {number} [fields.timestamp] - the starting timestamp of a span.
+     * @param {string} [fields.event] - a string message to be logged on a span.
+     * @param {string} [fields.payload] - an object to be logged on a span as a json string.
+     **/
+    log(fields: any): void {
+        if (this._spanContext.isSampled()) {
+            if (!fields.timestamp) {
+                fields.timestamp = Utils.getTimestampMicros();
+            }
+
+            if (!fields.event && !fields.payload) {
+                throw new Error('log must be passed either an event of type string, or a payload of type object');
+            }
+            let value = fields.event || JSON.stringify(fields.payload);
```

Comment:
I don't think payload should be used as the message in the log. I recently made a change in Go client to store payloads are `log_payload_#` tags: https://github.com/uber/jaeger-client-go/pull/31